### PR TITLE
refactor(module/loadbalancer): including examples

### PR DESCRIPTION
**Pull request by FoSix**
## Description

The Load Balancer module refactor

### CODE REVIEW

**Comment by FoSix** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.3 1.4 1.5 1.6"

**Comment by FoSix** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.3 1.4 1.5 1.6"

File: modules/loadbalancer/main.tf [hunk 1]
```diff
@@ -61,106 +58,148 @@ resource "azurerm_lb" "lb" {
...
+  probe_threshold     = each.value.probe_threshold
+  interval_in_seconds = each.value.interval_in_seconds
+  request_path        = each.value.protocol != "Tcp" ? each.value.request_path : null
+
+  # this is to overcome the discrepancy between the provider and Azure defaults
+  # for more details see here -> https://learn.microsoft.com/en-gb/azure/load-balancer/whats-new#known-issues:~:text=SNAT%20port%20exhaustion-,numberOfProbes,-%2C%20%22Unhealthy%20threshold%22
+  number_of_probes = 1
+}
+
+resource "azurerm_lb_rule" "this" {
```
Comment (sebastianczech) [inline modules/loadbalancer/main.tf:133]: Currently we have:
- `azurerm_lb_rule`
- `azurerm_lb_outbound_rule`

but we are missing `azurerm_lb_nat_rule` - what do you thing about adding support for that ?

**Review by sebastianczech — APPROVED** [review]
Overall great job 👍

File: modules/loadbalancer/main_test.go
[the remark this answers is not in the document]
Comment (FoSix) [reply modules/loadbalancer/main_test.go:1]: mea culpa, a mistake

File: modules/loadbalancer/main.tf (hunk 1 above)
Comment (FoSix) [reply modules/loadbalancer/main.tf:133]: we can, if there is a use case for it, with NGFW that is

File: modules/loadbalancer/variables.tf [hunk 2]
```diff
@@ -129,114 +158,286 @@ variable "frontend_ips" {
...
-  type        = string
+  type = map(object({
+    name                     = string
+    public_ip_name           = optional(string)
+    create_public_ip         = optional(bool, false)
+    public_ip_resource_group = optional(string)
+    subnet_id                = optional(string)
+    private_ip_address       = optional(string)
+    gwlb_fip_id              = optional(string)
+    in_rules = optional(map(object({
```
[the remark this answers is not in the document]
Comment (FoSix) [reply modules/loadbalancer/variables.tf:181]: @sebastianczech I've made [this example](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/f8cc1ca464757aa5338e6a04b8e89129f7ef5e7f/examples/lb/example.tfvars#L33-L86), IMO it's less readable - we still have a huge block of code

**Pull request merged**